### PR TITLE
allow report description to be localizable string or xpath expression

### PR DESCRIPTION
**Pull request by nickpell**
requested for https://trello.com/c/tqWMP6er/88-ui-for-reporting-to-match-other-existing-modules

@orangejenny cc: @NoahCarnahan 

screen shot 2015-11-23 at 7 01 27 pm

### CODE REVIEW

File: corehq/apps/app_manager/fixtures/mobile_ucr.py [hunk 1]
```diff
@@ -61,6 +61,8 @@ def _report_config_to_fixture(self, report_config, user):
         report_elem = ElementTree.Element('report', attrib={'id': report_config.uuid})
         report = ReportConfiguration.get(report_config.report_id)
         report_elem.append(self._element('name', localize(report_config.header, user.language)))
+        if not report_config.use_xpath_description:
+            report_elem.append(self._element('description', localize(report_config.localized_description, user.language)))
```
Comment (dimagimon) [inline corehq/apps/app_manager/fixtures/mobile_ucr.py]: E501 line too long (122 > 115 characters)

File: corehq/apps/app_manager/suite_xml/features/mobile_ucr.py [hunk 2]
```diff
@@ -163,7 +163,7 @@ def _column_to_series(column):
                         ),
                         template=Template(
                             text=Text(
-                                xpath=Xpath(function=config.description)
+                                xpath=Xpath(function=config.xpath_description if config.use_xpath_description else 'description')
```
Comment (dimagimon) [inline corehq/apps/app_manager/suite_xml/features/mobile_ucr.py]: E501 line too long (129 > 115 characters)

File: corehq/apps/app_manager/models.py [hunk 3]
```diff
@@ -3443,13 +3445,15 @@ def __init__(self, *args, **kwargs):
...
     @classmethod
     def wrap(cls, doc):
-        # for backwards compatibility with apps that have localized descriptions
-        from corehq.apps.userreports.util import default_language, localize
-        if isinstance(doc.get('description'), dict):
-            if doc['description']:
-                doc['description'] = localize(doc['description'], default_language())
-            else:
-                doc['description'] = ''
+        # for backwards compatibility with apps that have localized or xpath descriptions
```
Comment (orangejenny) [inline corehq/apps/app_manager/models.py]: How complex would it to be to do a non-lazy migration eventually? I'm assuming there aren't many of these docs in existence, and this code is getting a touch convoluted. Not in scope for this PR, just a thought for the future.

File: corehq/apps/app_manager/templates/app_manager/partials/mobile_report_configs.html [hunk 4]
```diff
@@ -60,7 +60,22 @@
                             <input type="text" data-bind="value: display">
                         </td>
                         <td>
-                            <input type="text" data-bind="value: description">
+                            <p>
```
Comment (orangejenny) [inline corehq/apps/app_manager/templates/app_manager/partials/mobile_report_configs.html]: move underneath text box please (it's a less common action, for more advanced users)

File: corehq/apps/app_manager/templates/app_manager/partials/mobile_report_configs.html [hunk 5]
```diff
@@ -60,7 +60,22 @@
...
+                                <input type="checkbox" data-bind="checked: useXpathDescription">
+                                {% trans "Use xpath expression" %}
+                            </p>
+                            <div data-bind="visible: useXpathDescription">
+                                <input type="text" data-bind="value: xpathDescription">
```
Comment (orangejenny) [inline corehq/apps/app_manager/templates/app_manager/partials/mobile_report_configs.html]: nice touch

File: corehq/apps/app_manager/templates/app_manager/partials/mobile_report_configs.html (hunk 4 above)
Comment (orangejenny) [inline corehq/apps/app_manager/templates/app_manager/partials/mobile_report_configs.html]: hmm, i see that'll move it under the warnings, too, that should be fine.

Comment (orangejenny) [inline corehq/apps/app_manager/templates/app_manager/partials/mobile_report_configs.html]: small tweak: changing this from a `p` to a `label` with class `checkbox` will a) align the box and the text and b) make the text clickable: http://getbootstrap.com/2.3.2/base-css.html#forms

File: corehq/apps/app_manager/tests/data/suite/reports_module_summary_detail_use_localized_description.xml [hunk 6]
```diff
@@ -31,7 +31,7 @@
         </header>
         <template>
           <text>
-            <xpath function="report description"/>
+            <xpath function='description'/>
```
Comment (orangejenny) [inline corehq/apps/app_manager/tests/data/suite/reports_module_summary_detail_use_localized_description.xml]: I'm not following the test changes: if this file is using a localized description, shouldn't it use a `locale` element for the description, not an `xpath` element?

File: corehq/apps/app_manager/models.py (hunk 3 above)
Comment (nickpell) [inline corehq/apps/app_manager/models.py]: I don't think it should be too complex - agree that this lazy migration code shouldn't hang around indefinitely.

File: corehq/apps/app_manager/tests/data/suite/reports_module_summary_detail_use_localized_description.xml (hunk 6 above)
Comment (nickpell) [inline corehq/apps/app_manager/tests/data/suite/reports_module_summary_detail_use_localized_description.xml]: previously the description was passed in the report fixture.  this reverts to that behavior when the localized description is used.

Comment (orangejenny) [inline corehq/apps/app_manager/tests/data/suite/reports_module_summary_detail_use_localized_description.xml]: aha, thanks

File: corehq/apps/app_manager/fixtures/mobile_ucr.py (hunk 1 above)
Comment (dimagimon) [inline corehq/apps/app_manager/fixtures/mobile_ucr.py]: E501 line too long (122 > 115 characters)

File: corehq/apps/app_manager/suite_xml/features/mobile_ucr.py (hunk 2 above)
Comment (dimagimon) [inline corehq/apps/app_manager/suite_xml/features/mobile_ucr.py]: E501 line too long (129 > 115 characters)

File: corehq/apps/app_manager/templates/app_manager/partials/mobile_report_configs.html (hunk 4 above)
Comment (nickpell) [inline corehq/apps/app_manager/templates/app_manager/partials/mobile_report_configs.html]: nice, updated.

**Comment by nickpell** [comment]
tests pass :white_check_mark:

**Comment by orangejenny** [comment]
do you disagree re: moving checkbox?

**Comment by nickpell** [comment]
sorry, misread your comment.  updated

**Pull request merged**